src/libstore/gc.cc

```diff
@@ -110,6 +110,10 @@ void LocalStore::createTempRootsFile()
 
 void LocalStore::addTempRoot(const StorePath & path)
 {
+    if (readOnly) {
+      debug("Read-only store doesn't support creating lock files for temp roots, but nothing can be deleted anyways.");
+      return;
+    }
```

Comment:
```suggestion
    }

```
since there are blank lines within the rest of the function's body, I think this makes it easier to read.